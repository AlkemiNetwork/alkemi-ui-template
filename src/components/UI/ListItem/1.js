import React from "react";
import PropTypes from "prop-types";

import { withStyles } from "@material-ui/core/styles";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";

const styles = theme => ({
    iconSelected: {
        color: theme.palette.primary.main
    }
});

const ListItem1 = ({ classes, icon, text, onClick, selected, ...props }) => (
    <ListItem button onClick={onClick} {...props}>
        <ListItemIcon className={selected ? classes.iconSelected : undefined}>
            {icon}
        </ListItemIcon>
        <ListItemText primary={text} />
    </ListItem>
);

ListItem1.propTypes = {
    icon: PropTypes.node.isRequired,
    text: PropTypes.string.isRequired,
    color: PropTypes.string,
    onClick: PropTypes.func.isRequired
};

export default withStyles(styles)(ListItem1);
