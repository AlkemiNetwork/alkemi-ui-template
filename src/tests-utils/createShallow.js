import { shallow as _shallow } from "enzyme";

const createShallow = ({ dive }) => {
    const shallow = (node, options) => {
        let wrapper = _shallow(node, options);

        for (let i = 0; i < dive; i++) {
            if (wrapper.name().match(/^WithWidth(.+)$/)) {
                // Diving through a Meterial-UI WidthWith component can not be performed
                // by default because of React.Fragment used in withWidth
                // It is a known issue: https://github.com/airbnb/enzyme/issues/1213
                wrapper = wrapper
                    .dive()
                    .children()
                    .at(0);
            } else {
                wrapper = wrapper.dive();
            }
        }
        return wrapper;
    };

    return shallow;
};

export default createShallow;
