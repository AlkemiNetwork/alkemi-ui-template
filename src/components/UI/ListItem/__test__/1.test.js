import React from "react";
import toJson from "enzyme-to-json";

import HomeIcon from "@material-ui/icons/Home";
import ListItem from "@material-ui/core/ListItem";
import ListItemIcon from "@material-ui/core/ListItemIcon";
import ListItemText from "@material-ui/core/ListItemText";
import getClasses from "@material-ui/core/test-utils/getClasses";

import ListItem1 from "../1";
import createShallow from "../../../../tests-utils/createShallow";

describe("<ListItem1 />", () => {
    let shallow;
    let wrapper;
    let classes;
    let props = {};

    beforeAll(() => {
        shallow = createShallow({ dive: 1 });
    });

    beforeEach(() => {
        props = {
            onClick: jest.fn(),
            text: "Test-Text",
            icon: <HomeIcon />
        };
        classes = classes = getClasses(<ListItem1 {...props} />);
        wrapper = shallow(<ListItem1 {...props} />);
    });

    it("renders 1 ListItem, ListItemIcon, ListItemText", () => {
        expect(wrapper.find(ListItem).length).toEqual(1);
        expect(wrapper.find(ListItemIcon).length).toEqual(1);
        expect(wrapper.find(ListItemText).length).toEqual(1);
    });

    it("renders 1 HomeIcon", () => {
        expect(wrapper.find(HomeIcon).length).toEqual(1);
    });

    it("ListItem button is functional", () => {
        wrapper
            .find(ListItem)
            .at(0)
            .simulate("click");
        expect(props.onClick.mock.calls.length).toEqual(1);
    });

    describe("when selected=true", () => {
        beforeEach(() => {
            wrapper = shallow(<ListItem1 {...props} selected />);
        });

        it("matches snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("ListItemIcon has class iconSelected", () => {
            expect(
                wrapper
                    .find(ListItemIcon)
                    .at(0)
                    .hasClass(classes.iconSelected)
            ).toBe(true);
        });
    });

    describe("when selected=false", () => {
        beforeEach(() => {
            wrapper = shallow(<ListItem1 {...props} selected={false} />);
        });

        it("matches snapshot", () => {
            expect(toJson(wrapper)).toMatchSnapshot();
        });

        it("ListItemIcon has not class iconSelected", () => {
            expect(
                wrapper
                    .find(ListItemIcon)
                    .at(0)
                    .hasClass(classes.iconSelected)
            ).toBe(false);
        });
    });
});
