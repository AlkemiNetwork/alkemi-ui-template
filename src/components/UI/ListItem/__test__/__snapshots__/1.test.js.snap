// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ListItem1 /> when selected=false matches snapshot 1`] = `
<WithStyles(ListItem)
  button={true}
  onClick={[MockFunction]}
>
  <WithStyles(ListItemIcon)>
    <pure(HomeIcon) />
  </WithStyles(ListItemIcon)>
  <WithStyles(ListItemText)
    primary="Test-Text"
  />
</WithStyles(ListItem)>
`;

exports[`<ListItem1 /> when selected=true matches snapshot 1`] = `
<WithStyles(ListItem)
  button={true}
  onClick={[MockFunction]}
>
  <WithStyles(ListItemIcon)
    className="ListItem1-iconSelected-4"
  >
    <pure(HomeIcon) />
  </WithStyles(ListItemIcon)>
  <WithStyles(ListItemText)
    primary="Test-Text"
  />
</WithStyles(ListItem)>
`;
